gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,351 @@
+      "Tous les êtres humains naissent libres et égaux en dignité et en droits.",
+      "Ils sont doués de raison et de conscience et doivent agir ",
+      "les uns envers les autres dans un esprit de fraternité.");
+
+  private static final Logger log = Logger.getLogger(ITGcsNio.class.getName());
+  private static final String BUCKET = RemoteGcsHelper.generateBucketName();
+  private static final String SML_FILE = "tmp-test-small-file.txt";
+  private static final int SML_SIZE = 100;
+  // it's big, relatively speaking.
+  private static final String BIG_FILE = "tmp-test-big-file.txt";
+  // arbitrary size that's not too round.
+  private static final int BIG_SIZE = 2 * 1024 * 1024 - 50;
+  private static final String PREFIX = "tmp-test-file";
+  private static Storage storage;
+  private static StorageOptions storageOptions;
+
+  private final Random rnd = new Random();
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    // loads the credentials from local disk as par README
+    RemoteGcsHelper gcsHelper = RemoteGcsHelper.create();
+    storageOptions = gcsHelper.options();
+    storage = storageOptions.service();
+    // create and populate test bucket
+    storage.create(BucketInfo.of(BUCKET));
+    fillFile(storage, SML_FILE, SML_SIZE);
+    fillFile(storage, BIG_FILE, BIG_SIZE);
+  }
+
+  @AfterClass
+  public static void afterClass() throws ExecutionException, InterruptedException {
+    if (storage != null && !RemoteGcsHelper.forceDelete(storage, BUCKET, 5, TimeUnit.SECONDS)) {
+      if (log.isLoggable(Level.WARNING)) {
+        log.log(Level.WARNING, "Deletion of bucket {0} timed out, bucket is not empty", BUCKET);
+      }
+    }
+  }
+
+  private static byte[] randomContents(int size) {
+    byte[] bytes = new byte[size];
+    new Random(size).nextBytes(bytes);
+    return bytes;
+  }
+
+  private static void fillFile(Storage storage, String fname, int size) throws IOException {
+    storage.create(BlobInfo.builder(BUCKET, fname).build(), randomContents(size));
+  }
+
+  @Test
+  public void testFileExists() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.exists(path)).isTrue();
+  }
+
+  @Test
+  public void testFileSize() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    assertThat(Files.size(path)).isEqualTo(SML_SIZE);
+  }
+
+  @Test(timeout=60_000)
+  public void testReadByteChannel() throws IOException {
+    CloudStorageFileSystem testBucket = getTestBucket();
+    Path path = testBucket.getPath(SML_FILE);
+    long size = Files.size(path);
+    SeekableByteChannel chan = Files.newByteChannel(path, StandardOpenOption.READ);
+    assertThat(chan.size()).isEqualTo(size);
+    ByteBuffer buf = ByteBuffer.allocate(SML_SIZE);
+    int read = 0;
+    while (chan.isOpen()) {
```

Comment:
In a collapsed comment from earlier, you mentioned that a loop was removed due to it only running once (since the size of the file was small, 100 bytes).  This loop seems similar; is it necessary?